### ISSUE TITLE
Assertion error message sequencing

### ISSUE THREAD

**Issue by ianbrandt** [assertj/assertj-core#1324]
I love AssertJ, but I find the sequencing of many of the error messages to be counterintuitive.

Take for example:

> Expecting: `<actual>` to be equal to: `<expected>` but was not.

[Per Wikipedia](https://en.wikipedia.org/wiki/Colon_(punctuation)):

> The most common use of the colon is to inform the reader that what follows the colon proves, explains, defines, describes, or lists elements of what preceded it.

Accordingly, when I begin reading that assertion error message my mind is, well, _expecting_ to see "Expecting: `<expected>`", not "Expecting: `<actual>`".  It's only when I read the closing, "but was not" that I realize I've interpreted the whole error message backwards.

I believe something like the following would be more intuitive and concise:

> Expected: `<expected>`, but was: `<actual>`.

I realize this may be a matter of personal preference, or of locale, so I'm filing this issue mainly to see if others find the current message style counterintuitive as well.

If enough do, maybe the solution would be to just change it, or perhaps some sort of localization or message style facility could be introduced to make it configurable.

**Comment by ianbrandt** [comment]
Some prior art for configurable reporting styles in Atrium:

https://github.com/robstoll/atrium/blob/v0.7.0/README.md#reporterbuilder.

**Comment by joel-costigliola** [comment]
The rationale of the error message design was to follow how the assertion read where actual comes in before expected, but you are raising a fair point.

I'm happy with the `Expecting: <expected>, but was: <actual>. ` suggestion or better with displaying both values on different for ease of comparison (when you don't have IDEs visual comparison). 

I think it only applies for `isEqualTo` though, for example `isBefore` error message looks like:
```
Expecting:
  <2011-01-01T00:00:00.000>
to be strictly before:
  <2012-01-01T00:00:00.000>
```
which seems fine to me.

Thoughts  on that latter error message ?

**Comment by martin-tarjanyi** [comment]
I also found it difficult to process that format of the error message when I started to use the library. I'm still not completely get used to it (after a couple of months of usage).

For the latter I would prefer something like this:
```
Actual:
  <2011-01-01T00:00:00.000>
is not strictly before:
  <2012-01-01T00:00:00.000>
```
Although, I agree, it is somewhat easier to process in the original format, than the original message of `isEqualTo`.
